armory/scenarios/dapricot_scenario.py

```diff
@@ -0,0 +1,174 @@
+            except NotImplementedError:
+                raise NotImplementedError(
+                    "Training has not yet been implemented for object detectors"
+                )
+
+        if defense_type == "Transform":
+            # NOTE: Transform currently not supported
+            logger.info(f"Transforming estimator with {defense_type} defense...")
+            defense = load_defense_wrapper(config["defense"], estimator)
+            estimator = defense()
+
+        metrics_logger = metrics.MetricsLogger.from_config(
+            config["metric"], skip_benign=True, skip_attack=False, targeted=True,
+        )
+
+        eval_split = config["dataset"].get("eval_split", "test")
+
+        # Evaluate the ART estimator on adversarial test examples
+        logger.info("Generating or loading / testing adversarial examples...")
+
+        attack = load_attack(attack_config, estimator)
+        test_data = load_dataset(
+            config["dataset"],
+            epochs=1,
+            split=eval_split,
+            num_batches=num_eval_batches,
+            shuffle_files=False,
+        )
+
+        export_samples = config["scenario"].get("export_samples")
+        if export_samples is not None and export_samples > 0:
+            sample_exporter = SampleExporter(
+                self.scenario_output_dir, test_data.context, export_samples
+            )
+        else:
+            sample_exporter = None
+
+        for x, y in tqdm(test_data, desc="Attack"):
+            with metrics.resource_context(
+                name="Attack",
+                profiler=config["metric"].get("profiler_type"),
+                computational_resource_dict=metrics_logger.computational_resource_dict,
+            ):
+
+                if x.shape[0] != 1:
+                    raise ValueError("D-APRICOT batch size must be set to 1")
+                # (nb=1, num_cameras, h, w, c) --> (num_cameras, h, w, c)
+                images = x[0]
+                y_object, y_patch_metadata = y
+
+                generate_kwargs = deepcopy(attack_config.get("generate_kwargs", {}))
+                generate_kwargs["y_patch_metadata"] = y_patch_metadata
+                y_target = label_targeter.generate(y_object)
+                generate_kwargs["y_object"] = y_target
+
+                x_adv = attack.generate(x=images, **generate_kwargs)
+
+            # Ensure that input sample isn't overwritten by estimator
+            x_adv.flags.writeable = False
+            y_pred_adv = estimator.predict(x_adv)
+            for img_idx in range(len(y_object)):
+                y_i_target = y_target[img_idx]
+                y_i_pred = y_pred_adv[img_idx]
+                metrics_logger.update_task(
+                    [y_i_target], [y_i_pred], adversarial=True, targeted=True
+                )
+
+            metrics_logger.update_perturbation(x, x_adv)
+            if sample_exporter is not None:
+                sample_exporter.export(x, x_adv, y, y_pred_adv)
```

Comment:
sample_exporter.export(x[0], x_adv, y, y_pred_adv)

Reply:
Not sure about the cleanest logic to save the correct number of samples since sample_exporter.export treats each input as a batch of 3 for the dapricot_dev_adversarial dataset. Perhaps a small note in docs/scenarios.md would suffice?